### DISCUSSION TITLE
proc-macro is explicitly disabled from async_trait not ignored

### DISCUSSION THREAD

**Discussion by lbennett-stacki** [mrcjkb/rustaceanvim#482] [Q&A]
I am using rustaceanvim with lazyvim.

I see that the rustaceanvim config has proc macros enabled and async-traits ignored:

```
{
	"mrcjkb/rustaceanvim",
	version = "^4",
	ft = { "rust" },
	opts = {
		server = {
			...,
			default_settings = {
				-- rust-analyzer language server configuration
				["rust-analyzer"] = {
					cargo = {
						allFeatures = true,
						loadOutDirsFromCheck = true,
						buildScripts = {
							enable = true,
						},
					},
					-- Add clippy lints for Rust.
					checkOnSave = true,
					procMacro = {
						enable = true,
						ignored = {
							["async-trait"] = { "async_trait" },
							["napi-derive"] = { "napi" },
							["async-recursion"] = { "async_recursion" },
						},
					},
				},
			},
		},
	},
        ...
}
```

But I am still getting this diagnostic message..

![Screenshot 2024-08-18 at 15 03 34](https://github.com/user-attachments/assets/8fb032b4-526f-461d-a075-0cc8766a8166)

**Comment by mrcjkb** [comment]
Hey 👋 

This is a LazyVim issue. Please ask there 😀
I don't have the capacity to support all the Neovim distributions out there.

**Comment by unkcpz** [comment]
Came across the same issue and here is the the workaround from LazyVim https://github.com/LazyVim/LazyVim/discussions/5638
May it can be documented somewhere in rustaceanvim? Anyway, I assume this link can help people with same issue quickly find the solution.

**Reply by mrcjkb** [reply]
> May it can be documented somewhere in rustaceanvim? 

I don't have the capacity to support every 3rd party Neovim distro out there, so no.
